lib/rules/no-restricted-imports.js

```diff
@@ -344,6 +389,27 @@ module.exports = {
 						restrictedPathEntry.importNames;
 					const allowedImportNames =
 						restrictedPathEntry.allowImportNames;
+					const allowTypeImports =
+						restrictedPathEntry.allowTypeImports;
+
+					// Skip if this is a type-only import and it's allowed for this specific entry
+					if (
+						(node.type === "ImportDeclaration" ||
+							node.type === "TSImportEqualsDeclaration") &&
+						isTypeOnlyImport(node) &&
```

Comment:
Also not sure about this one, since both rules report it. Adding an allowed value import makes an otherwise allowed type import not allowed:

```js
/* eslint no-restricted-imports: ["error", {  
  "paths": [{
    "name": "foo",
    "importNames": ["bar"],
    "allowTypeImports": true
  }]
}]
*/

/* eslint @typescript-eslint/no-restricted-imports: ["error", {  
  "paths": [{
    "name": "foo",
    "importNames": ["bar"],
    "allowTypeImports": true
  }]
}]
*/

import { type bar } from "foo"; // this is valid

import { type bar, baz } from "foo"; // but this reports (both rules): 'bar' import from 'foo' is restricted

```

Reply:
Fixed in https://github.com/eslint/eslint/pull/19562/commits/a5adab3cd128abb48b1d892da5bb21a721e0b942